### ISSUE TITLE
User data storage

### ISSUE THREAD

**Issue by ss-o** [z-shell/api#6]
# User data storage

I currently considering storing user data in r2 as requests can be easily made zi, web and etc. This issue is for any related topic.

- [R2](https://developers.cloudflare.com/r2/get-started/)

**Comment by 0xMRTT** [comment]
It means that we need to maintain an account for each user. So maybe your idea is to create zshhub (like GitHub or dockerhub)?

**Comment by 0xMRTT** [comment]
The first things to do is to imagine the architecture and how everything can work together. We can make a microservice for each thing?

**Comment by ss-o** [comment]
> It means that we need to maintain an account for each user. So maybe your idea is to create zshhub (like GitHub or dockerhub)?

Not exactly, consider it just as a component for the API which can be used on any trigger.

**Comment by 0xMRTT** [comment]
> > The first things to do is to imagine the architecture and how everything can work together. We can make a microservice for each thing?
> 
> Without visualizing (maybe it is only me) I can't create anything :octocat:

It's not only you 😃. I want to create a diagram

**Comment by ss-o** [comment]
I like https://app.diagrams.net/

**Comment by 0xMRTT** [comment]
Oh sorry
